ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -466,50 +521,66 @@ static int
 gcp_object_list(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *directory,
   const char *prefix, char ***object_list, uint32_t *count)
 {
-    WT_UNUSED(file_system);
-    WT_UNUSED(session);
-    WT_UNUSED(directory);
-    WT_UNUSED(prefix);
-    WT_UNUSED(object_list);
-    WT_UNUSED(count);
-
-    return 0;
+    return gcp_object_list_helper(
+      file_system, session, directory, prefix, object_list, count, false);
 }
 
+// Allocate and initialize an array of C strings from vector of strings.
+//
+// Requires count <= objects.list().
+// count==0 is valid, and in this case object_list will be set to nullptr.
+// Caller is responsible for memory allocated for object_list.
 static int
-gcp_object_list_add(const gcp_store &gcp_, char ***object_list,
-  const std::vector<std::string> &objects, const uint32_t count)
+make_object_list(char ***object_list, const std::vector<std::string> &objects, const uint32_t count)
 {
-    WT_UNUSED(gcp_);
-    WT_UNUSED(object_list);
-    WT_UNUSED(objects);
-    WT_UNUSED(count);
+    if (count < 1) {
```

Comment:
We can directly check for 0 here since its an unsigned int for better readability

Reply:
I disagree: the advantage of this test is that it is true regardless of signed-ness of count.

Reply:
I see your point, I would like to get a third perspective onto this, @tammybailey what are your thoughts?

Reply:
My preference is also for `count == 0`, and I agree that we shouldn't be casually changing the signedness of `count`.

Reply:
[2247e06](https://github.com/wiredtiger/wiredtiger/pull/8794/commits/2247e061d343aef915a73a5310503cefd4c82e0e)